diff --git a/meta-luv/classes/luv-efi.bbclass b/meta-luv/classes/luv-efi.bbclass
index 6e72d4813f..e5b83b2199 100644
--- a/meta-luv/classes/luv-efi.bbclass
+++ b/meta-luv/classes/luv-efi.bbclass
@@ -310,7 +310,11 @@ python build_efi_cfg() {
 
     if re.search("aarch64", target):
        cfgfile.write('menuentry \'sbbr/sbsa\' {\n')
-       cfgfile.write('chainloader /EFI/BOOT/Shell.efi\n')
+       cfgfile.write('set oldroot=$root\n')
+       cfgfile.write('set root=(tftp,${PXE_IMAGE_SERVER})\n')
+       cfgfile.write('chainloader EFI/BOOT/sbsa/ShellSbsa.efi\n')
+       cfgfile.write('set root=$oldroot\n')
+       cfgfile.write('boot\n')
        cfgfile.write('}\n')
 
     cfgfile.close()
@@ -357,6 +361,10 @@ python build_luv_cfg() {
     comment = 'Add test suites if needed, to run reboot tests in LUV'
     name = 'LUVCFG_reboot_tests'
     luvcfg.write(insert_var(name, comment, d))
+
+    comment = 'The IP Address of the Server where the image of PXE boot is stored'
+    name = 'PXE_image_server'
+    luvcfg.write(insert_var(name, comment, d))
 
     luvcfg.write('\n#####################################################\n')
     luvcfg.write('## User should not modify any parameter below this ##\n')
diff --git a/meta-luv/recipes-core/images/luv-image.inc b/meta-luv/recipes-core/images/luv-image.inc
index 62b55d1d26..93bdd50bbe 100644
--- a/meta-luv/recipes-core/images/luv-image.inc
+++ b/meta-luv/recipes-core/images/luv-image.inc
@@ -36,6 +36,8 @@ LUVCFG_tests_append_x86-64 = "bits,chipsec,ndctl'"
 LUVCFG_tests_append_aarch64 = "'"
 LUVCFG_reboot_tests = "LUV_REBOOT_TESTS=none"
 LUVCFG_bits_can_run = "LUV_BITS_CAN_RUN='true'"
+PXE_image_server = "PXE_IMAGE_SERVER=none"
+
 
 # Path to splash screen image
 FILESEXTRAPATHS_append := "${THISDIR}/luv-live-image:"
